src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -2139,6 +2139,17 @@ private function addMailerSection(ArrayNodeDefinition $rootNode, callable $enabl
                                 ->end()
                             ->end()
                         ->end()
+                        ->arrayNode('webhook')
+                            ->info('Mailer Webhook configuration')
+                            ->normalizeKeys(false)
+                            ->useAttributeAsKey('provider')
+                            ->prototype('array')
+                                ->normalizeKeys(false)
+                                ->useAttributeAsKey('name')
+                                ->requiresAtLeastOneElement()
+                                ->prototype('scalar')->end()
```

Comment:
agreed, but I wanted to make the configuration not directly depended on the sendgrid bridge. so I didn't exactly know how to do it the right way, this was just a way so that the sendgrid bridge (and more in the feature) have the ability to define a config value. 

any correct suggestions?

Reply:
If the processing of the configuration in the extension is specific, making the Configuration class generic is actually worse as it means that both classes have a different understanding of the structure of the configuration.